lib/sse.js

```diff
@@ -0,0 +1,301 @@
+'use strict'
+
+const _ = require('lodash')
+const mongojs = require('mongojs')
+const Promise = require('bluebird')
+const hl = require('highland')
+const Boom = require('boom')
+
+/*
+ Usage:
+ ======================================
+ When setting up Multi SSE (ie: SSE for multiple resources), you just need to pass in Harvester context as such:
+
+ this.multiSSE = new SSE()
+ this.multiSSE.init({
+ context: server
+ })
+
+ You can then point an EventReader to listen from "{base_url}/changes/streaming?resouces=foo,bar,baz".
+
+ When setting up SSE for a single route, you will need to pass the resource name:
+
+ this.singleSSE = new SSE()
+ this.singleSSE.init({
+ singleResourceName: 'foo'
+ })
+
+ You can then point an EventReader to listen from "{base_url}/foo/changes/streaming".
+
+ Verbs:
+ ======================================
+ You can also pass a "verbs" option to this module. If none is passed, SSE will only listen to "insert" events from uplog.
+ Values you can pass are "post", "put" and "delete" which in turn currespond to oplog ops "i", "u" and "d".
+
+ this.singleSSE = new SSE()
+ this.singleSSE.init({
+ context: server,
+ singleResourceName: 'foo',
+ verbs: ['post', 'put', 'delete']
+ })
+ */
+
+var SSE = function () {
+}
+
+SSE.prototype.init = function (config) {
```

Comment:
Prototypes are so ```2008``` in ES6 apparently. Can we use the same style of module as others?